### PR TITLE
Feature oe sep glue 70201

### PR DESCRIPTION
**Pull request by saimekala07**

### CODE REVIEW

**Review by TreyE — APPROVED** [review]
Looks correct, thanks for making fixes.

File: app/models/external_events/enrollment_event_notification.rb [hunk 1]
```diff
@@ -418,6 +418,31 @@ def already_processed_termination?
...
+        return false unless pol.employer_id == find_employer(policy_cv).id
+      end
+      return false if pol.canceled?
+      return false if pol.terminated?
+      return false if pol.subscriber.blank?
+      return false if pol.subscriber.m_id != subscriber_id
+      return false if pol.plan.blank?
+      return false unless existing_plan.carrier_id == pol.plan.carrier_id
+      return false unless existing_plan.coverage_type == pol.plan.coverage_type
+      return false unless existing_plan.year + 1 == pol.plan.year
```
Comment (TreyE) [inline app/models/external_events/enrollment_event_notification.rb:442]: Simply checking if it's for next year won't be enough.  You have to also check if the coverage would have been contiguous.  If it wouldn't have been, i.e. imagine they don't have a 2021 renewal, but a policy which begins Feb 1, 2021, we should not terminate.

File: app/models/external_events/enrollment_event_notification.rb [hunk 2]
```diff
@@ -418,6 +418,31 @@ def already_processed_termination?
       end
     end
 
+    def policies_to_cancel
```
Comment (TreyE) [inline app/models/external_events/enrollment_event_notification.rb:421]: Can we name the method something more like - renewal_policies_to_cancel?  I want to be able to tell that it is doing this on a renewal, and I think the method would be more descriptive of its function.

**Review by TreyE — CHANGES_REQUESTED** [review]
Please see my comments.  They apply likewise I believe for both 88680 and 88683.

**Review by TreyE — APPROVED** [review]
Looks correct.

**Review by TreyE — APPROVED** [review]

**Review by bill-transue — APPROVED** [review]
Looks good, this will call `termination_cancels_renewal` for CareFirst cancellations and if it is a renewal cancellation it will call `cancel_via_hbx!` when active coverage is terminated.

**Review by bill-transue — APPROVED** [review]
This will silently cancel the previous carrier's auto renewal when a consumer switches carriers.

**Pull request merged**